pkg/filesystem/local_directory_windows.go

```diff
@@ -0,0 +1,261 @@
+// +build windows
+package filesystem
+
+import (
+	"errors"
+	"internal/syscall/windows"
+	"log"
+	"path"
+	"path/filepath"
+	"os"
+	"sort"
+	"strings"
+	"syscall"
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// Windows does not differentiate permissions for users, group and others.
+	// Therefore, a mapping table is needed.
+	FileMode644 = 0666
+	FileMode666 = 0666
+	FileMode700 = 0777
+	DirMode700 = 0777
+	DirMode777 = 0777
+	SymLinkMode777 = 0666
+)
+
+const uncPrefix = `\\?\`
+
+type localDirectory struct {
+	longAbsPath string
+}
+
+func validateFilename(name string) error {
+	if name == "" || name == "." || name == ".." || strings.ContainsRune(name, '/') {
+		return status.Errorf(codes.InvalidArgument, "Invalid filename: %#v", name)
+	}
+	return nil
+}
+
+func PortableSyscallUmask(mask int) () {
+	// No implementation on Windows
+}
+
+// NewLocalDirectory creates a directory handle that corresponds to a
+// local path on the system.
+func NewLocalDirectory(path string) (Directory, error) {
+	path = strings.ReplaceAll(path, "/", string(os.PathSeparator))
+	longAbsPath, err := filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+	if !strings.HasPrefix(longAbsPath, uncPrefix) {
+		longAbsPath = uncPrefix + longAbsPath
+	}
+	return newLocalDirectoryLongAbsPath(longAbsPath)
+}
+
+func newLocalDirectoryLongAbsPath(longAbsPath string) (Directory, error) {
+	fileInfo, err := os.Lstat(longAbsPath)
+	if err != nil {
+		return nil, err
+	}
+	if !fileInfo.IsDir() {
+		return nil, syscall.ENOTDIR
+	}
+	d := &localDirectory{
+		longAbsPath: longAbsPath,
+	}
+	return d, nil
+}
+
+func (d *localDirectory) validateFilenameAndJoin(name string) (string, error) {
+	if err := validateFilename(name); err != nil {
+		return "", err
+	}
+	// path.Join uses the wrong path separator, which is critical for UNC paths. Join manually.
```

Comment:
What about `filepath.Join()`?

Reply:
Thank you for pointing it out! (This is my first Go code ever, so I need help to find the right APIs.)